src/shogun/mathematics/linalg/LinalgBackendEigen.h

```diff
@@ -30,20 +30,23 @@
  * Authors: 2016 Pan Deng, Soumyajit De, Heiko Strathmann, Viktor Gal
  */
 
+//#include <shogun/lib/SGVector.h>
 #include <shogun/mathematics/eigen3.h>
 #include <shogun/mathematics/linalg/LinalgBackendBase.h>
 
-#ifndef Linalg_Backend_Eigen_H__
-#define Linalg_Backend_Eigen_H__
+#ifndef LINALG_BACKEND_EIGEN_H__
+#define LINALG_BACKEND_EIGEN_H__
 
 namespace shogun
 {
 
-/** @brief linalg methods with EIGEN3 backend */
+/** @brief linalg methods with Eigen3 backend
```

Comment:
No need to reference base class here as doxygen does that already automatixcally